common/beerocks/bwl/CMakeLists.txt

```diff
@@ -12,46 +12,49 @@ file(GLOB_RECURSE bwl_common_sources ${MODULE_PATH}/common/*.c*)
 # Support more permissive c++ language
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
 
-# Default BWL Type
+# Default BWL Type and Libraries
 set(BWL_TYPE "DUMMY")
-
-# UGW
-if (TARGET_PLATFORM STREQUAL "ugw")
-
-    set(BWL_TYPE "DWPAL")
-    # Extra libraries
-    list(APPEND BWL_LIBS rt fapiwlancommon dl helper ugwhelper blobmsg_json)
-    # UGW platform libraries
-    link_directories(${PLATFORM_STAGING_DIR}/usr/lib)
-    # Extra libraries (for dwpal)
-    link_directories(${PLATFORM_STAGING_DIR}/opt/lantiq/lib)
-    # hostap dir
-    set(HOSTAPD_DIR "${PLATFORM_BUILD_DIR}/iwlwav-hostap-2.6")
-    
-# RDKB
-elseif(TARGET_PLATFORM STREQUAL "rdkb")
-
-    set(BWL_TYPE "DWPAL")
-    # Extra libraries
-    list(APPEND BWL_LIBS rt dl)
-    # Extre directories
-    include_directories(${PLATFORM_INCLUDE_DIR}/wav-dpal)
+list(APPEND BWL_LIBS rt dl)
 
 # Linux
-elseif(TARGET_PLATFORM STREQUAL "linux")
-    set(BWL_TYPE "DUMMY")
+if(TARGET_PLATFORM STREQUAL "linux")
     set(BEEROCKS_TMP_PATH "/tmp/$ENV{USER}/beerocks")
     add_definitions(-DBEEROCKS_TMP_PATH="${BEEROCKS_TMP_PATH}")
 
-# Turris Omnia
-elseif(TARGET_PLATFORM STREQUAL "turris-omnia")
+# OpenWRT
+elseif (TARGET_PLATFORM STREQUAL "openwrt")
+
+    # UGW
+    if (TARGET_PLATFORM_TYPE STREQUAL "ugw")
+        set(BWL_TYPE "DWPAL")
 
-    set(BWL_TYPE "NL80211")
+        # dwpal search path
+        link_directories(${PLATFORM_STAGING_DIR}/opt/intel/lib)
```

Comment:
Can we not use `link_directories` and instead add a `Finddwpal.cmake` and use `find_package` like we do for other libraries such as `safec`?